File: js/src/2264-largestGoodInteger.js
/**
 * [2264] 字符串中最大的 3 位相同数字
 * @lc app=leetcode.cn id=2264 lang=javascript
 * @param {string} num
 * @return {string}
 */
const largestGoodInteger = function(num) {
  let tmp = '';
  for (let i = 2; i < num.length; i++) {
    if (num[i] === num[i - 1] && num[i] === num[i - 2]) {
      if (!tmp || Number(num[i]) > Number(tmp)) {
        tmp = num[i];
      }
    }
  }
  if (tmp === '') return '';
  return tmp + tmp + tmp;
};

export { largestGoodInteger };

File: ruby/src/2264-largest-good-integer.rb
# @param {String} num
# @return {String}
def largest_good_integer(num)
  tmp = ''

  for i in 2...(num.length)
    if num[i] === num[i - 1] && num[i] === num[i - 2]
      # string.to_i or string.to_f
      if tmp.length == 0 || (num[i].to_i > tmp.to_i)
        tmp = num[i];
      end
    end
  end

  if tmp === ''
    return '';
  end
  
  return tmp + tmp + tmp;  
end
